actionpack/lib/action_controller/renderer.rb

```diff
@@ -0,0 +1,60 @@
+module ActionController
+  class Renderer
+    class_attribute :controller, :defaults
+    attr_reader :env
+
+    class << self
+      delegate :render, to: :new
+
+      def for(controller)
+        Class.new self do
+          self.controller = controller
+          self.defaults = {
```

Comment:
Let's add config/initializers/application_controller_renderer.rb that shows how to use this on setup. Specifically the way to toggle https depending on the env and setting the http_host.